docs/cookbook/entities/custom-model-accessible-per-admin-channel.rst

```diff
@@ -0,0 +1,392 @@
+
+.. code-block:: yaml
+
+    sylius_plus:
+    permissions:
+        app_admin_supplier_index:
+            parent: suppliers
+            label: action.index
+        app_admin_supplier_create:
+            parent: suppliers
+            label: action.create
+        app_admin_supplier_update:
+            parent: suppliers
+            label: action.update
+        app_admin_supplier_delete:
+            parent: suppliers
+            label: action.delete
+        app_admin_supplier_show:
+            parent: suppliers
+            label: action.show
+        app_admin_supplier_bulk_delete:
+            parent: suppliers
+            label: action.bulk_delete
+
+.. note::
+
+    More information about using administrator roles (ACL/RBAC) can be found :doc:`here </book/customers/admin_user>`.
+
+* Create ``App\Checker\ResourceChannelEnabilibityChecker`` and decorate ``Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelEnabilibityCheckerInterface``.
+
+* Then add ``Supplier`` as checking resource:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Checker;
+
+    use Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelEnabilibityCheckerInterface;
+    use Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelEnabilibityChecker as DecoratedResourceChannelEnabilibityChecker;
+
+    final class ResourceChannelEnabilibityChecker implements ResourceChannelEnabilibityCheckerInterface
+    {
+        /** @var DecoratedResourceChannelEnabilibityChecker */
+        private $decoratedResourceChannelEnabilibityChecker;
+
+        public function __construct(DecoratedResourceChannelEnabilibityChecker $decoratedResourceChannelEnabilibityChecker)
+        {
+            $this->decoratedResourceChannelEnabilibityChecker = $decoratedResourceChannelEnabilibityChecker;
+        }
+
+        public function forResourceName(string $resourceName): bool
+        {
+            if (!$this->decoratedResourceChannelEnabilibityChecker->forResourceName($resourceName)) {
+                return in_array($resourceName, [
+                    'supplier'
+                ]);
+            }
+
+            return true;
+        }
+    }
+
+.. code-block:: yaml
+
+    App\Checker\ResourceChannelEnabilibityChecker:
+        decorates: Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelEnabilibityCheckerInterface
+        arguments: ['@.inner']
+
+* Create ``App\Checker\ResourceChannelChecker`` and decorate ``Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelCheckerInterface`` next add condition for checking ``Supplier``.
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Checker;
+
+    use Sylius\Plus\ChannelAdmin\Application\Checker\ResourceChannelCheckerInterface;
+    use Sylius\Plus\Entity\ChannelInterface;
+
+    final class ResourceChannelChecker implements ResourceChannelCheckerInterface
```

Comment:
We definitely should at least define in `ResourceChannelChecker`, in Plus the behaviour for checking the entity with `ChannelAwareInterface` and then recommend for implementing this interface instead of decorating this service